worker/download.go

```diff
@@ -505,7 +505,10 @@ func (mgr *downloadManager) newSlabDownload(slice object.SlabSlice, migration bo
 	// build sector info
 	hostToSectors := make(map[types.PublicKey][]sectorInfo)
 	for sI, s := range slice.Shards {
-		hostToSectors[s.LatestHost] = append(hostToSectors[s.LatestHost], sectorInfo{s, sI})
+		si := sectorInfo{s, sI}
+		for hk := range s.Contracts {
```

Comment:
I would still try to give preference to the `LatestHost` here

Reply:
I noticed this issue here https://github.com/SiaFoundation/renterd/issues/1590 so I'll go ahead with the removal. I kept the comment on the PR because `LatestHost` was giving us extra information, which for me was pretty valuable information since the `LatestHost` is the most reliable (or "best") in theory.